core/android/src/main/kotlin/com/walletconnect/android/internal/common/signing/cacao/Cacao.kt

```diff
@@ -104,25 +95,52 @@ fun Cacao.Payload.toCAIP222Message(chainName: String = "Ethereum"): String {
     return message
 }
 
-private fun Cacao.Payload.getActionsString(issuer: Issuer): String {
-    return decodeReCaps(issuer).entries.joinToString(", ") { "'${it.key}': " + it.value.joinToString(", ") { value -> "'$value'" } }
+private fun Cacao.Payload.getActionsString(): String {
+    val map = decodeReCaps()
+    if (map.isEmpty()) throw Exception("Decoded ReCaps map is empty")
+    var result = ""
+    var index = 1
+
+    map.forEach { (key, values) ->
```

Comment:
Look into using reduce instead. It's optimized to do what you're trying to do here. Something like this


```suggestion
    map.reduceIndexed { (index, result: String, mapEntry: Map.Entry<*, *>) ->
```